daml-lf/interface/src/main/scala/com/digitalasset/daml/lf/iface/reader/InterfaceReader.scala

```diff
@@ -170,6 +175,23 @@ object InterfaceReader {
       (k, tyVars.toSeq, Variant(fields.toSeq))
     }
 
+  private def foldEnums(
+      state: State,
+      a: (InterfaceReaderError.Tree, Iterable[(QualifiedName, ImmArraySeq[Ref.Name], Enum)]))
+    : State =
+    addPartitionToState(state, a) {
+      case (st, (k, typVars, enum)) => st.addEnum(k, typVars, enum)
+    }
+
+  private[reader] def enum(m: ModuleName)(a: DamlLf1.DefDataType)
+    : InterfaceReaderError.Tree \/ (QualifiedName, ImmArraySeq[Ref.Name], Enum) =
+    (locate('name, rootErrOf[ErrorLoc](fullName(m, a.getName))).validation |@|
+      locate('typeParams, typeParams(a)).validation |@|
+      locate('constructors, rootErrOf[ErrorLoc](constructors(a))).validation) {
+      (k, tyVars, constructors) =>
+        (k, tyVars.toSeq, Enum(constructors.toSeq))
```

Comment:
`(_:ImmArraySeq[...]).toImmArray.toSeq` = `identity`

```suggestion
        (k, tyVars.toSeq, Enum(constructors))
```

Reply:
done